src/backend/core/user/authentication.py

```diff
@@ -0,0 +1,19 @@
+from .models import User
+from django.contrib.auth.backends import BaseBackend
+
+
+class UserAuthBackend(BaseBackend):
+    def authenticate(self, request, username=None, password=None):
+        try:
+            user = User.objects.get(email=username)
+            if user.check_password(password):
+                return user
+            return None
+        except User.DoesNotExist:
+            return None
+
+    def get_user(self, user_id):
+        try:
+            return User.objects.get(pk=user_id)
```

Comment:
This is just an authentication backend. Its called when somebody uses [django.auth.authenticate](https://docs.djangoproject.com/en/4.1/topics/auth/default/#django.contrib.auth.authenticate) function. Workflow you are describing should be reflected in some of the views implementation (currently its not, so it will be added)